data/scripts/hackmd2cyberblog.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+"""
+    A script to convert HackMD notes to Cyber Blog posts (mostly downloading and replacing images).
+
+    Dependencies:
+        You will need to download the `requests` library to run this script.
+
+        pip install requests
+
+    Usage: hackmd2cyberblog.py <file>
+
+    The script will download all images from the HackMD note and replace the URLs with the local paths.
+    
+    NOTE: Make sure to run this from the root of the repository (uses relative paths).
+
+    Example:
+        python3 data/scripts/hackmd2cyberblog.py data/blog/2024-12-03-fall-2024-fuzzing-lab.md
+"""
+
+import sys, os, re
+import requests
+
+if len(sys.argv) < 2:
+    print("Usage: hackmd2cyberblog.py <file>")
+    sys.exit(1)
+
+# Read the file
+with open(sys.argv[1], "r") as f:
+    content = f.read()
+
+# Return a list of all image urls
+images = re.findall(r"https://hackmd\.io/_uploads/[^)]+", content)
+
+
+def download_image(dowload_folder: str, url: str) -> str:
+    """
+    Download an image from a given URL and return the path to the downloaded image.
+    """
+    response = requests.get(url)
+    if response.status_code != 200:
+        print(f"[*] Failed to download image: {url}")
+        return None
+
+    # Save the image
+    filename = url.split("/")[-1]
+    path = os.path.join(dowload_folder, filename)
+    print(f"[*] Downloading image: {url} -> {path}")
```

Comment:
I think we should print before starting the download rather than at the end, and have a different message like "Image X saved at filepath" or whatever

Reply:
Due the the amount of comments, let me split the script to a separate PR so it doesn't block this one.